### PR TITLE
`JSON_ARRAY` correctly handles `CHAR` bindvar params

### PR DESCRIPTION
**Pull request by fulghum**
The `JSON_ARRAY` function wasn't able to handle a `[]byte` literal expression in a bind var. I also noticed that although our script tests can specify `Bindings`, they weren't getting used by our prepared query runners. I also fixed that, so that I could add a new test for this issue. 

Fixes: https://github.com/dolthub/go-mysql-server/issues/1855

Related Vitess change: https://github.com/dolthub/vitess/pull/261

Dolt CI Checks: https://github.com/dolthub/dolt/pull/6441

### CODE REVIEW

File: sql/expression/function/json_array.go [hunk 1]
```diff
@@ -111,7 +111,10 @@ func (j *JSONArray) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 			return nil, err
 		}
 
-		if json, ok := val.(types.JSONValue); ok {
+		if bytes, ok := val.([]byte); ok {
```
Comment (JCOR11599) [inline sql/expression/function/json_array.go:114]: I think this has a side-effect of converting all `BINARY` / `BLOB` values into strings.

Some test cases and output from MySQL:
```sql
mysql> create table t (b blob);
mysql> insert into t values ("abc");
mysql> select json_array(b) from t;
+-------------------------+
| json_array(b)           |
+-------------------------+
| ["base64:type252:YWJj"] |
+-------------------------+
1 row in set (0.0005 sec)
```

```sql
mysql> select json_array(cast('123abc' as binary));
+--------------------------------------+
| json_array(cast('123abc' as binary)) |
+--------------------------------------+
| ["base64:type15:MTIzYWJj"]           |
+--------------------------------------+
1 row in set (0.0004 sec)
```

```sql
mysql> select json_contains(json_array(cast('blob' as binary)), json_array('blob'), '$');
+----------------------------------------------------------------------------+
| json_contains(json_array(cast('blob' as binary)), json_array('blob'), '$') |
+----------------------------------------------------------------------------+
|                                                                          0 |
+----------------------------------------------------------------------------+
1 row in set (0.0004 sec)

mysql> select json_contains(json_array(cast('blob' as binary)), json_array(cast('blob' as binary)), '$');
+--------------------------------------------------------------------------------------------+
| json_contains(json_array(cast('blob' as binary)), json_array(cast('blob' as binary)), '$') |
+--------------------------------------------------------------------------------------------+
|                                                                                          1 |
+--------------------------------------------------------------------------------------------+
1 row in set (0.0004 sec)
```

Comment (JCOR11599) [reply sql/expression/function/json_array.go:114]: Also maybe using a `switch case` is cleaner

**Review by JCOR11599 — COMMENTED** [review]
Prepared and bindvar improvements are great, but have a correctness side effect

File: sql/expression/function/json_array.go (hunk 1 above)
Comment (fulghum) [reply sql/expression/function/json_array.go]: Good feedback on the interaction with valid binary use cases for `JSON_ARRAY`. Unfortunately there didn't seem to be a good way to distinguish the type information for those bind vars. I dug into the vitess code a little deeper to see what's going on, and sure enough, the caller **does** send type information for that param and marks it as `CHAR`, but Vitess swallows that info for some reason and sends it back up to GMS as `VARBINARY`. That doesn't seem correct to drop that type info from the caller, so I've got a Vitess PR (https://github.com/dolthub/vitess/pull/261) that I'm testing out that preserves that type info. That change also lets me drop the `[]byte` → `string` conversion, so if it works, that should address your concern with the valid binary use cases.

**Review by JCOR11599 — APPROVED** [review]
LGTM!

**Pull request merged**